### ISSUE TITLE
Dismiss Click on "... Files Changed" Label When "Discard All Changes" Option is Opened

### ISSUE THREAD

**Issue by JosephTLyons** [desktop/desktop#1858]
### Description

When you right click on the "... Files Changed" Label under the Changes tab, it prompts you with the option to "Discard All Changes."  If you want to dismiss this message, you just click the whitespace around it, but if you accidentally click the label itself, it will select or deselect all files.  Would it be possible to turn off clicks to that "... Files Changed" label when the "Discard All Changes" box is opened?  Its not too big of a deal to move the mouse to another part of the screen, but sometimes, I'm not actively thinking about where to click to dismiss the message.  This may be a bug however, because I notice other clickable elements aren't triggered when you use their surface as a clicking space to dismiss the "Discard All Changes" option.
### Version

**GitHub Desktop version:** [0.5.9]

**OS version:** [macOS 10.12.5]

**Comment by shiftkey** [comment]
> If you want to dismiss this message, you just click the whitespace around it, but if you accidentally click the label itself, it will select or deselect all files. 

This feels like The Bug™, and I can reproduce it over 50% of the time - here's a quick and dirty GIF of it:

![](https://cloud.githubusercontent.com/assets/359239/26679047/cdf50040-46a9-11e7-853b-377e5d24ef06.gif)

**Comment by JosephTLyons** [comment]
Yep, this is exactly it.  Strange though, on my system it happens every time.

**Comment by shiftkey** [comment]
@JosephTLyons I think this is related to the click behaviour of the header, and maybe we're not handling the "dismiss context menu" correctly [in here](https://github.com/desktop/desktop/blob/master/app/src/ui/changes/changes-list.tsx#L104-L116).

**Comment by JosephTLyons** [comment]
@shiftkey  - boy I sure wish I could help, but I'm unfamiliar with this language entirely.

**Comment by thegiantbeast** [comment]
I have been poking around on the code to see where this problem is coming from and it seems that this issue only happens after the first "click" event.

To explain, if I "right-click" and click anywhere (even on the empty spaces or on top of the label text) the `onContextMenu` is fired just fine but as soon as I click on the header to toggle the checkbox it starts to call `onContextMenu` when "right-click" and `onClick` when "left-click". [and yes, I've added a `onClick` listener to double check this issue]

Hope this makes sense to someone because I couldn't figure out where the issue is coming from.
If I put a `event.preventDefault()` inside the `onClick` it works (of course).
So I tried to workaround it by setting a flag for when the menu is visible, etc but no luck (and I also know that this wouldn't be the final fix, I was just messing around/getting to know the codebase).
